apps/src/componentLibrary/dropdown/_CustomDropdown.tsx

```diff
@@ -125,38 +155,47 @@ const CustomDropdown: React.FunctionComponent<CustomDropdownProps> = ({
       ref={dropdownRef}
       aria-describedby={ariaProps['aria-describedby']}
     >
-      <button
-        id={`${name}-dropdown-button`}
-        type="button"
-        className={moduleStyles.dropdownButton}
-        data-toggle="dropdown"
-        onClick={toggleDropdown}
-        disabled={disabled}
-        {...ariaProps}
-        aria-haspopup={true}
-        aria-label={ariaProps['aria-label'] || `${name} filter dropdown`}
-      >
-        {isSomeValueSelected && (
-          <FontAwesomeV6Icon iconName="check-circle" iconStyle="solid" />
-        )}
-        {icon && (
-          <FontAwesomeV6Icon
-            iconName={icon.iconName}
-            iconStyle={icon.iconStyle}
-            title={icon.title}
-            className={icon.className}
-          />
-        )}
-        <span
-          className={classNames(
-            moduleStyles.dropdownLabel,
-            moduleStyles[`dropdownLabel-${labelType}`]
-          )}
+      {useDSCOButtonAsTrigger ? (
+        <Button
+          {...triggerComponentProps}
+          {...triggerButtonProps}
```

Comment:
Do we need some validation to check that if `useDSCOButtonAsTrigger` is true, then `triggerButtonProps` must be defined?

Reply:
Button component wil throw an error by itslef in case anytheng will go wrong, though agree makes sense to add some validation in `_CustomDropdown` as well, to emphasise that triggerButtonProps must be defined. (Main reason - right now we don't pass `text` and/or `icon`(s) to Button via `triggerComponentProps` which are required for `Button`)